x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/Subject.java

```diff
@@ -133,12 +133,12 @@ public Type getType() {
         final RoleReference.ApiKeyRoleReference limitedByRoleReference = new RoleReference.ApiKeyRoleReference(
             apiKeyId,
             limitedByRoleDescriptorsBytes,
-            "apikey_limited_role"
+            true
         );
         if (isEmptyRoleDescriptorsBytes(roleDescriptorsBytes)) {
             return List.of(limitedByRoleReference);
         }
-        return List.of(new RoleReference.ApiKeyRoleReference(apiKeyId, roleDescriptorsBytes, "apikey_role"), limitedByRoleReference);
+        return List.of(new RoleReference.ApiKeyRoleReference(apiKeyId, roleDescriptorsBytes, false), limitedByRoleReference);
```

Comment:
It's probably a _nit_, but I think an enum would be better than a `boolean`. Something like:
```
enum ApiKeyRoleType {
  ASSIGNED_ROLES, LIMITED_BY_ROLES
}
```
(I'm happy with different names for them enum class, or members)

Reply:
I added the enum and also use the enum toString for the source of the roleKey to avoid having extra string constants. It might also become useful if later we decide to support layered limited-by. I think it's an overall improvement. Thanks for the suggestion.